src/SerialFirmata.cpp

```diff
@@ -66,6 +66,11 @@ boolean SerialFirmata::handleSysex(byte command, byte argc, byte *argv)
                 // because all Arduino pins are set to OUTPUT by default in StandardFirmata.
                 pinMode(pins.rx, INPUT);
               }
+#if defined(SERIAL_STORE_AND_FORWARD)    
```

Comment:
added
```C
    if (portId < SERIAL_READ_ARR_LEN)
    {
...
    }
  }
  return false;
```
.. and SERIAL_READ_ARR_LEN is "only" 12, not even the expected 16, but it is the right choise and enough for 4 software serial ports